### ISSUE TITLE
Newbie! How to use

### ISSUE THREAD

**Issue by kylepritchard** [thechampagne/webview-d#1]
I am new to D and would like to convert an existing project which uses webview to D. I am wondering how to use this library, specifically linking. I have tried compiling webview to object file and linking but cannot get it to work. Do you have any basic guidelines?

**Comment by thechampagne** [comment]
In [basic](https://github.com/thechampagne/webview-d/tree/main/examples/basic) you can find working example that link to libweview.so.

**Comment by thechampagne** [comment]
If you want to link with object instead of library you need to place the object file in source directory and change
```
libs "webview"
```
to
```
sourceFiles "source/webview.o"
libs "stdc++" "gtk+-3.0" "webkit2gtk-4.0"
```
In dub.sdl.

**Comment by kylepritchard** [comment]
I managed to get it working just before your reply. Using windows PC so was able to include the library locally and link through dub. 

I am struggling with binding functions to the webview instance. Do you have any examples of how that can be done?

**Comment by thechampagne** [comment]
Do you mean how to use this function ?
```d
void bind(string name, BindCallback fn, void* arg);
```

**Comment by kylepritchard** [comment]
Yes, i cant ever seem to get the signature of the function correct and I believe a context is required whilst program is running according to webview docs

**Comment by thechampagne** [comment]
```d
extern(C) void callBack(string seq, string req, void* arg)
{
 // code
}

w.bind("name", &callBack, null);

**Comment by gitgit3** [comment]
I didn't find this thread until after I got webview working. I did things differently. After downloading the webview project, I ran the build.sh in webview/script. This built webview.o as well as libwebview.so in the webview/build/ directory.

Then, since I an using dub.json, I appended the following property which allows the linker to find the lib:
"libs": ["$PACKAGE_DIR/libs/webview/build/library/libwebview.so"]

It seems like $PACKAGE_DIR is your project root where the dub.json file is located.

Hope that helps someone!
